src/xwayland/xwm/mod.rs

```diff
@@ -2401,3 +2401,31 @@ fn send_selection_notify_resp(
     conn.flush()?;
     Ok(())
 }
+
+fn send_configure_notify(
+    conn: &RustConnection,
+    win: &X11Window,
+    geometry: Rectangle<i32, Logical>,
+    override_redirect: bool,
+) -> Result<(), ConnectionError> {
+    conn.send_event(
+        false,
+        *win,
+        EventMask::STRUCTURE_NOTIFY,
+        ConfigureNotifyEvent {
+            response_type: CONFIGURE_NOTIFY_EVENT,
+            sequence: 0,
+            event: *win,
+            window: *win,
```

Comment:
No idea how X works, is assigning window id to event field intentional?

Reply:
Yes. https://gitlab.freedesktop.org/wayland/weston/-/blob/main/xwayland/window-manager.c#L725-726